Capstone Pokemon Project

yersel500 opened the pull request:
We implement the following features:

- The home page.
- The comments popup.
- We followed the layout of the wireframes provided. 
- When the Home page loads, the web app retrieves data from:
    - The selected API shows the list of items on the screen.
    - The Involvement API to show the item likes.
- Our page  make only 2 requests:
    - One to the base API.
    - And one to the Involvement API.
- When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.
- When the user clicks on the "Comments" button, the Comments popup appears.
- When the popup loads, the web app retrieves data from the selected API and shows details about the selected item.
- The Involvement API to show the item comments.
 - When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

You can see a live demo video here:  [See the video](https://drive.google.com/file/d/1MToAuQClMnZuT64QwpIiKB4A1V7XS1ME/view?usp=sharing)

estebanmual committed:
On branch feature/set-webpack
	new file:   .eslintrc.json
	new file:   .gitignore
	new file:   .hintrc
	new file:   .stylelintrc.json
	new file:   linters.yml
	new file:   package-lock.json
	new file:   package.json
	new file:   src/app/app.js
	new file:   src/app/utils/files.md
	new file:   src/assets/assets.md
	new file:   src/index.js
	new file:   src/main.css
	new file:   src/template.html
	new file:   src/vendor.js
	new file:   webpack.common.js
	new file:   webpack.dev.js
	new file:   webpack.prod.js
	Set up webpack and linters.

estebanmual committed:
On branch feature/set-webpack
	new file:   .github/workflows/linters.yml
	Add linters.yml file to show them in github

estebanmual committed:
linters errors

estebanmual committed:
linters errors

estebanmual committed:
Merge pull request #22 from yersel500/feature/set-webpack

 On branch feature/set-webpack

yersel500 committed:
Setup jest

yersel500 committed:
Fix linters errors

yersel500 committed:
Set up bootstrap

yersel500 committed:
Fix linters errors

yersel500 committed:
Merge pull request #23 from yersel500/set-jest

Set up jest

estebanmual committed:
On branch feature/header
	new file:   dist/assets/pokemon-logo.9dc16171f2cf9c91f9c2..png
	new file:   dist/index.html
	new file:   dist/main.31d6cfe0d16ae931b73c.js
	new file:   dist/main.b7c02203ed454aef3e76.css
	new file:   dist/vendor.f3d9b71d948e40bcb814.js
	new file:   dist/vendor.f3d9b71d948e40bcb814.js.LICENSE.txt
	deleted:    src/assets/assets.md
	new file:   src/assets/pokemon-logo.png
	modified:   src/template.html
	Added the header and the footer

yersel500 committed:
Merge pull request #24 from yersel500/feature/header

 On branch feature/header

yersel500 committed:
get data from api

yersel500 committed:
Fix display home

yersel500 committed:
Improve cards style

yersel500 committed:
Fix linters errors

yersel500 committed:
Fix eslint errors

yersel500 committed:
Clean code

yersel500 committed:
Merge pull request #25 from yersel500/display-homepage

Display homepage

estebanmual committed:
On branch feature/comments-popup
	deleted:    src/app/utils/files.md
	modified:   src/app/utils/loadpokemons.js
	modified:   src/app/utils/showpokemons.js
	Added popup in which you can see the img of the pokemon. and some Data about it. Also added basis style of the comments area

estebanmual committed:
On branch feature/comments-popup
	new file:   src/app/utils/getComments.js
	modified:   src/app/utils/loadpokemons.js
	new file:   src/app/utils/postComments.js
	modified:   src/app/utils/showpokemons.js
	Added functionality to:
	 - Display comments pop up
	 - Display commments of item
	 - Create a new comment
	 - Add comment counter

estebanmual committed:
Merge pull request #26 from yersel500/feature/comments-popup

Comments pop up

yersel500 committed:
Change pokemon images

yersel500 committed:
Add event listener to like image

yersel500 committed:
Show updated likes

yersel500 committed:
Style couter items

yersel500 committed:
Add test for number of items

yersel500 committed:
Fix Linter errors

yersel500 committed:
Merge pull request #29 from yersel500/new-like

Add likes and test counter

estebanmual committed:
On branch feature/comments-test
	modified:   README.md
	new file:   dist/assets/heart.1af03272208329adeeaf..svg
	modified:   dist/index.html
	renamed:    dist/main.b7c02203ed454aef3e76.css -> dist/main.49336dc5a5db9aee151a.css
	new file:   dist/main.633f9c98ff3a415ca974.js
	deleted:    dist/main.a2622470534c5ce83895.js
	modified:   src/app/app.js
	modified:   src/app/utils/counterItems.js
	modified:   src/app/utils/getLikes.js
	modified:   src/app/utils/loadpokemons.js
	modified:   src/app/utils/postLikes.js
	modified:   src/app/utils/showpokemons.js
	deleted:    src/assets/heart2.svg
	deleted:    src/assets/smallheart.png
	modified:   src/main.scss
	modified:   src/template.html
	new file:   src/tests/countComments.js
	new file:   src/tests/countComments.test.js
	renamed:    src/app/utils/countItems.js -> src/tests/countItems.js
	renamed:    src/app/utils/countItems.test.js -> src/tests/countItems.test.js
	I added funciton to display automatically the comment once you commented. Also added the test for the comments counter.

estebanmual committed:
Merge pull request #30 from yersel500/feature/comments-test

 Comments Test

estebanmual committed:
Solve likes issue

Gambit142 commented on README.md:
```diff
@@ -1 +1,46 @@
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
```
- [x] (Optional) It would be nice to create an _**Acknowledgement Section**_ that mentions the Pokemon and Involvement API used for this project. 👍

Gambit142 commented on src/app/utils/counterItems.js:
```diff
@@ -0,0 +1,8 @@
+export default function counterItems(array) {
+  const counter = array.length;
+  const counterPok = document.querySelector('.counterItems');
+  const counterNav = document.querySelector('.counterPokemons');
+  counterPok.innerHTML = `Choose between our ${counter} pokemons`;
+  counterNav.innerHTML = `(${counter})`;
+  return counter;
+}
```
- [x] Kindly use `Es6` syntax for creating a function because it was specified in the project that you should use `Es6`. It would look something like this:

`export default const functionName = (parameters) => {code in here...}`

Gambit142 commented on src/app/utils/getComments.js:
```diff
@@ -0,0 +1,7 @@
+export default async function getComments(id) {
+  const itemUrl = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/FBTV7LVQNoUbR7Qxl6jA/comments?item_id=${id}`;
+  const response = await fetch(itemUrl);
+  const data = await response.json();
+  return data;
+  // this returns comment, username and creation_date
+}
```
- [x] - [ ] Kindly use `Es6` syntax for creating a function because it was specified in the project that you should use `Es6`. It would look something like this:

`export default const functionName = (parameters) => {code in here...}`

Gambit142 commented on src/app/utils/showpokemons.js:
```diff
@@ -0,0 +1,115 @@
+        const itemID = e.target.id;
+        const commmentsData = Array.from(await getComments(itemID));
+        const commentCounter = commmentsData.length;
+        if (commentCounter !== 0) {
+          const commentsSection = document.querySelector(`.displayCommentsSection-${itemID}`);
+          commentsSection.innerHTML = `
+          <h5>Comments (${commentCounter})</h5> 
+        `;
```
- [ ] I noticed that when you add a comment, the `commentCounter` is not updated immediately until I close the popup and then reopen the same popup. What you want to do is to increase the `commentCounter` by 1 when someone successfully creates a comment or you immediately fetch all comments and display the length.

Gambit142 reviewed (changes requested):
Hi @yersel500 @estebanmual 👋, 

Good job so far!

### Highlights

- Good UI and design. ✔️ 
- All linters checks passed ✔️ 
- Video was recorded professionally ✔️ 
- Dry Code ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights (Changes)
- Kindly use `Es6` syntax to create all your functions. 👍 

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider the comment I made about the acknowledgement section of your `README.md` file.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

yersel500 committed:
Update README

estebanmual committed:
On branch develop
	modified:   src/app/utils/counterItems.js
	modified:   src/app/utils/getComments.js
	modified:   src/app/utils/showpokemons.js
	#
	I change the functions to Es6 syntax.
	Solve problem of not updating the comment counter

estebanmual committed:
Merge branch 'develop' of github.com:yersel500/capstone-js into develop

estebanmual committed:
Update the dist directory

estebanmual committed:
Update README.md

Meltrust reviewed (approved):
Hi @yersel500,


## Wow, you did it 🎉
![Brilliant](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)
## Thank you for making your functions use arrow syntax, for making the comments update immediately when added, and for creating an acknowledgments section as requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
